pytest_postgresql/factories/process.py

```
@@ -41,18 +41,71 @@
+#: ask for pg_ctl.exe by name. Running it does not, CreateProcess appends the extension.
+PG_CTL_NAMES = ("pg_ctl.exe", "pg_ctl") if platform.system() == "Windows" else ("pg_ctl",)
+
+
+def _pg_bindir() -> str | None:
+    """Ask pg_config where PostgreSQL keeps its binaries, None if it can't tell us.
+
+    pg_config comes from the client development package, which on most distributions
+    can be installed without the server, so its answer is a hint rather than an answer.
+    """
+    try:
+        return subprocess.check_output(["pg_config", "--bindir"], universal_newlines=True).strip()
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)